### PR TITLE
Fix/mpt

### PR DESCRIPTION
**Pull request by ethyla**

### CODE REVIEW

File: scripts/deployers/DeployFix.s.sol [hunk 1]
```diff
@@ -22,98 +22,26 @@ contract DeployFix is Script {
     address rootChainManager;
     // // pending check
 
-    // owned by 0xfa7d2a996ac6350f4b56c043112da0366a59b74c
```
Comment (dev1644) [inline scripts/deployers/DeployFix.s.sol:25]: Do we want to merge all these edits of DeployFix.s.sol  @ethyla ?

**Review by dev1644 — COMMENTED** [review]
LGTM!

File: forge/MerklePatriciaProof.sol [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                        (prefix == bytes1(uint8(2)) || prefix == bytes1(uint8(3)))
+                    ) {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+
+                //extension node
+                if (traversed == 0) {
```
Comment (web3security) [inline forge/MerklePatriciaProof.sol:94]: I would suggest we use the latest version here for the extension node prefix check.

File: forge/MerklePatriciaProof.sol [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                    return false;
+                }
+
+                pathPtr += traversed;
+                nodeKey = bytes32(RLPReader.toUintStrict(currentNodeList[1]));
+                // Alternative 1 end
```
Comment (web3security) [inline forge/MerklePatriciaProof.sol:100]: I would remove all "alternative" options (comments)

**Review by web3security — APPROVED** [review]
LGTM thanks!

**Pull request merged**